src/battle.ts

```diff
@@ -1153,7 +1160,7 @@ class Battle {
 		for (const side of this.sides) {
 			if (side) side.reset();
 		}
-		this.myPokemon = null;
+		this.mySide.myPokemon = null;
```

Comment:
Wait, what's the reason `myPokemon` was moved? I think it's fine where it was.

Reply:
Also I don't think there's a need to reset `myPokemon`.

Reply:
Oh, I see! Because your ally also has Pokémon. I would actually put it in `battle.myAllyPokemon`.

Reply:
The problem with putting it in `.mySide.myPokemon` is that it's very brittle. It gets cleared on `resetStep` and a lot of other stuff.